Mathlib/Order/RelSeries.lean

```diff
@@ -154,6 +157,125 @@ protected noncomputable def withLength [r.InfiniteDimensional] (n : ℕ) : RelSe
 lemma nonempty_of_infiniteDimensional [r.InfiniteDimensional] : Nonempty α :=
   ⟨RelSeries.withLength r 0 0⟩
 
+instance membership : Membership α (RelSeries r) :=
+  ⟨(· ∈ Set.range ·)⟩
+
+theorem mem_def {x : α} {s : RelSeries r} : x ∈ s ↔ x ∈ Set.range s :=
+  Iff.rfl
+
+/-- Start of a series, i.e. for `a₀ -r-> a₁ -r-> ... -r-> aₙ`, its head is `a₀`.
+
+Since a relation series is assumed to be non-empty, this is well defined. -/
+def head (x : RelSeries r) : α := x 0
+
+/-- End of a series, i.e. for `a₀ -r-> a₁ -r-> ... -r-> aₙ`, its last element is `aₙ`.  Since a
+relation series is assumed to be non-empty, this is well defined. -/
```

Comment:
```suggestion
/-- Start of a series, i.e. for `a₀ -r→ a₁ -r→ ... -r→ aₙ`, its head is `a₀`.

Since a relation series is assumed to be non-empty, this is well defined. -/
def head (x : RelSeries r) : α := x 0

/-- End of a series, i.e. for `a₀ -r→ a₁ -r→ ... -r→ aₙ`, its last element is `aₙ`.  Since a
relation series is assumed to be non-empty, this is well defined. -/
```